mockhandler.go

```diff
@@ -45,3 +63,20 @@ func ToJSON(obj interface{}) []byte {
 	}
 	return data
 }
+
+func SingleHeaderMatcher(key, value string) interface{} {
+	headers := make(http.Header)
+	headers.Set(key, value)
+	return MultiHeaderMatcher(headers)
+}
+
+func MultiHeaderMatcher(desiredHeaders http.Header) interface{} {
```

Comment:
I think httpmock knows how to compare map contents, so this function is only needed when we want to check for a particular subset of headers to make sure that if they exist then they're the right value, right? Figure we may want to call this out when you write the docs for these 2 matchers.

Reply:
yeah sounds good, the straight up map matcher would be hard because the golang client sends headers that we don't want to have to account for, but making sure documenting that this is matching a subset and ensuring they exist with the given values sound like a good move